backend/routes/shops.js

```diff
@@ -887,26 +811,23 @@ module.exports = function (router) {
       const additionalOpts = {}
 
       // Configure Stripe webhooks
-      if (IS_PROD) {
-        // Register webhooks only on prod
-        if (req.body.stripe === false) {
-          log.info(`Shop ${shopId} - Deregistering Stripe webhook`)
-          await deregisterStripeWebhooks(existingConfig)
-          additionalOpts.stripeWebhookSecret = ''
-          additionalOpts.stripeBackend = ''
-        } else if (req.body.stripe && !req.body.stripeWebhookSecret) {
-          log.info(`Shop ${shopId} - Registering Stripe webhook`)
-          const { secret } = await registerStripeWebhooks(
-            req.body,
-            existingConfig,
-            netConfig.backendUrl
-          )
-          additionalOpts.stripeWebhookSecret = secret
-        } else if (req.body.stripeWebhookSecret) {
-          additionalOpts.stripeWebhookSecret = req.body.stripeWebhookSecret
-        }
-      } else {
-        additionalOpts.stripeWebhookSecret = req.body.stripeWebhookSecret
+      if (req.body.stripe === false) {
+        log.info(`Shop ${shopId} - Deregistering Stripe webhook`)
+        await stripeUtils.deregisterWebhooks(req.shop, existingConfig)
+        additionalOpts.stripeWebhookSecret = ''
+        additionalOpts.stripeWebhookHost = ''
+        additionalOpts.stripeBackend = ''
+      } else if (req.body.stripe) {
+        log.info(`Shop ${shopId} - Registering Stripe webhook`)
+        const { secret } = await stripeUtils.registerWebhooks(
+          req.shop,
+          req.body,
```

Comment:
Should we pick which fields from the body are sent to Stripe?

And also before making the call to Stripe, should we double check the fields are valid? We have util methods to do so here: https://github.com/OriginProtocol/dshop/blob/master/packages/utils/stripe.js
I understand the UI may already do this validation but imho it's worth double checking.

Reply:
Didn't use `pick` since I deserialise the object in that function. Will add the validation